apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -2346,8 +2357,53 @@ func interceptorHandler(r *ReconcileAPI, instance *wso2v1alpha1.API, owner []met
 		if errorUpdateinterceptConf != nil {
 			log.Error(errorUpdateinterceptConf, "error in updating interceptors config map with owner reference")
 		}
-		return true, jobVolumeMount, jobVolume, nil
+		exsistBalInterceptors = true
+		errBalInterceptor = nil
+	}
+
+	//handle java interceptors
+	var confNames = instance.Spec.Definition.Interceptors.Java
+	if len(confNames) > 0 {
+		log.Info("java interceptor configmaps specified in API spec")
+		for _, configmapName := range confNames {
+			javaConfigmap, err := getConfigmap(r, configmapName, userNameSpace)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// Interceptor is not defined
+					log.Info("interceptor" + configmapName + " is not defined")
+					return exsistBalInterceptors, false, jobVolumeMount, jobVolume, errBalInterceptor, nil
+				} else {
+					// Error getting interceptors configmap.
+					log.Error(err, "error retrieving configmap " + configmapName)
+					return exsistBalInterceptors, false, jobVolumeMount, jobVolume, errBalInterceptor, err
+				}
+			} else if err == nil {
+				jobVolume = append(jobVolume, corev1.Volume{
+					Name: javaInterceptorsVolume,
```

Comment:
Can we have same name for many volumes (since in a foreach)?

Reply:
No it is one volume and multiple jar files will append to a one volume